options

maxiride opened the pull request:
Addresses #7 

Honestly I don't know how to handle an optional struct altogether, so there is a margin of improvement there either make it a variadic parameter or a pointer or something else if you want to keep consistency of it being an optional parameter.

maxiride committed:
options

r0l1 commented on fillpdf.go:
```diff
@@ -32,9 +32,17 @@ import (
 
+// Options represents the options to alter the PDF filling process
+// Overwrite will overwrite any pre existing filled PDF
+// Flatten will flatten the document making the form field no longer editable
```
Move the comments to their definitions. Each options field will have it's own comment...

maxiride replied:
Done.

r0l1 commented on fillpdf.go:
```diff
@@ -32,9 +32,17 @@ import (
-// One variadic boolean specifies, whenever to overwrite the destination file if it exists.
-func Fill(form Form, formPDFFile, destPDFFile string, overwrite ...bool) (err error) {
+// The options parameter alters few aspects of the generation.
+func Fill(form Form, formPDFFile, destPDFFile string, options Options) (err error) {
```
Make options variadic, to keep in sync with the previous API.

```go
var o Options
if len(options) > 0 {
  o = options[0]
}
```

maxiride replied:
Implemented default values and made the parameter variadic again.

r0l1 reviewed (commented):
Looks good. Let's stick to the variadic style. Pointer is not required for such a small struct.

maxiride committed:
default options and variadict parameters

Set default options values and makes the options parameters variadic (optional)
